urllib3/util/connection.py

```diff
@@ -99,3 +106,39 @@ def _set_socket_options(sock, options):
 
     for opt in options:
         sock.setsockopt(*opt)
+
+
+def family_filter():
```

Comment:
This name is really tricky, because it's contextual.

`get_address_family()` feels wrong because this function will *never* return `AF_INET6`. It has exactly two return values, `AF_INET` and `0`, where `0` is only meaningful when you realise that it gets passed into `getaddrinfo`. If the code returned `AF_UNSPEC` instead of `0` that would get us closer to something sensible.

Maybe return `AF_UNSPEC` and change the name to indicate that this is the family flag for `getaddrinfo` in some way: `gai_family`, for example? I dunno, just some thoughts.